### PR TITLE
PRM-823 Updated summary card to display text

### PR DESCRIPTION
**Pull request by ruthdas03**

### CODE REVIEW

File: app/viewmodels/LateSubmissionPenaltySummaryCard.scala [hunk 1]
```diff
@@ -31,5 +32,6 @@ case class LateSubmissionPenaltySummaryCard(
                         isAddedPoint: Boolean = false,
                         isAppealedPoint: Boolean = false,
                         appealStatus: Option[AppealStatusEnum.Value] = None,
-                        isAdjustedPoint: Boolean = false
+                        isAdjustedPoint: Boolean = false,
+                        multiplePenaltyPeriod:Option[Html] = None
```
Comment (LewisMitchell) [inline app/viewmodels/LateSubmissionPenaltySummaryCard.scala:36]:
```suggestion
                        multiplePenaltyPeriod: Option[Html] = None
```

File: app/viewmodels/SummaryCardHelper.scala [hunk 2]
```diff
@@ -422,4 +424,9 @@ class SummaryCardHelper @Inject()(link: views.html.components.link) extends Impl
     case _ => renderTag(messages("status.due"), "penalty-due-tag")
   }
 
+  private def getMultiplePenaltyPeriodMessage(penalty : PenaltyPoint)(implicit messages: Messages):Option[Html]={
```
Comment (LewisMitchell) [inline app/viewmodels/SummaryCardHelper.scala:427]:
```suggestion
  private def getMultiplePenaltyPeriodMessage(penalty: PenaltyPoint)(implicit messages: Messages): Option[Html] = {
```

File: app/viewmodels/SummaryCardHelper.scala [hunk 3]
```diff
@@ -422,4 +424,9 @@ class SummaryCardHelper @Inject()(link: views.html.components.link) extends Impl
...
+    if(penalty.period.getOrElse(Seq.empty).size > 1)
+    Some(Html(messages("lsp.multiple.penaltyPeriod",dateTimeToString(PenaltyPeriodHelper.sortedPenaltyPeriod(penalty.period.get).reverse.head.submission.dueDate))))
```
Comment (LewisMitchell) [inline app/viewmodels/SummaryCardHelper.scala:429]: This is some quite complex logic, could you sort it and then do .last:
```suggestion
    Some(Html(messages("lsp.multiple.penaltyPeriod", dateTimeToString(PenaltyPeriodHelper.sortedPenaltyPeriod(penalty.period.get).last.submission.dueDate))))
```

File: conf/messages [hunk 4]
```diff
@@ -63,6 +63,8 @@ lsp.onThreshold.p2.b2 = you’ll have to pay a £200 penalty every time you subm
 agent.lsp.onThreshold.p2.b2 = they’ll have to pay a £200 penalty every time they submit a VAT Return late
 lsp.onThreshold.link = Show me how to bring this account up to date
 
+lsp.multiple.penaltyPeriod = The VAT Return due on {0} was also submitted late. <br> HMRC  only applies 1 penalty for late submission in each month.
```
Comment (LewisMitchell) [inline conf/messages:66]:
```suggestion
lsp.multiple.penaltyPeriod = The VAT Return due on {0} was also submitted late. <br> HMRC only applies 1 penalty for late submission in each month.
```

File: conf/messages.cy [hunk 5]
```diff
@@ -63,6 +63,8 @@ lsp.onThreshold.p2.b2 = you’ll have to pay a £200 penalty every time you subm
 agent.lsp.onThreshold.p2.b2 = they’ll have to pay a £200 penalty every time they submit a VAT Return late
 lsp.onThreshold.link = Show me how to bring this account up to date
 
+lsp.multiple.penaltyPeriod = The VAT Return due on {0} was also submitted late. <br> HMRC  only applies 1 penalty for late submission in each month.
```
Comment (LewisMitchell) [inline conf/messages.cy:66]:
```suggestion
lsp.multiple.penaltyPeriod = The VAT Return due on {0} was also submitted late. <br> HMRC only applies 1 penalty for late submission in each month.
```

File: test/assets/messages/IndexMessages.scala [hunk 6]
```diff
@@ -122,4 +122,5 @@ object IndexMessages {
   val appealPointText = "Appeal this penalty"
   val checkAppeal = "Check if you can appeal"
   val appealStatus = "Appeal status"
+  def lspMultiplePenaltyPeriodMessage(dueDate : String) = s"The VAT Return due on $dueDate was also submitted late. <br> HMRC  only applies 1 penalty for late submission in each month."
```
Comment (LewisMitchell) [inline test/assets/messages/IndexMessages.scala:125]:
```suggestion
  def lspMultiplePenaltyPeriodMessage(dueDate: String) = s"The VAT Return due on $dueDate was also submitted late. <br> HMRC only applies 1 penalty for late submission in each month."
```

File: test/viewmodels/SummaryCardHelperSpec.scala [hunk 7]
```diff
@@ -300,6 +301,32 @@ class SummaryCardHelperSpec extends SpecBase with ImplicitDateFormatter {
         result.appealStatus.isDefined shouldBe true
         result.appealStatus.get shouldBe AppealStatusEnum.Tribunal_Rejected
       }
+
+      "show message for multiple penalty period " in {
+
+        val sampleSummaryCardReturnSubmitted: LateSubmissionPenaltySummaryCard = LateSubmissionPenaltySummaryCard(
```
Comment (LewisMitchell) [inline test/viewmodels/SummaryCardHelperSpec.scala:307]: You could rename this to expectedResult

File: test/views/components/LateSubmissionPenaltySummaryCardSpec.scala [hunk 8]
```diff
@@ -599,6 +669,27 @@ class LateSubmissionPenaltySummaryCardSpec extends SpecBase with ViewBehaviours
         docWithAppealedPointUnderTribunalRejected.select("dd").get(4).text() shouldBe "Appeal rejected by tax tribunal Read outcome message"
       }
     }
+
+    "given multiple penalty period in LSP" should{
```
Comment (LewisMitchell) [inline test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:673]:
```suggestion
    "given multiple penalty period in LSP" should {
```

File: test/views/components/LateSubmissionPenaltySummaryCardSpec.scala [hunk 9]
```diff
@@ -599,6 +669,27 @@ class LateSubmissionPenaltySummaryCardSpec extends SpecBase with ViewBehaviours
...
+      implicit val doc: Document = asDocument(summaryCardHtml.apply(summaryCardModelWithMultiplePenaltyPeriodLSP))
+      "show message VAT return submitted earlier in multiple penalty period" in{
+        doc.select("p.govuk-body").text() shouldBe "The VAT Return due on 24 May 2021 was also submitted late. HMRC only applies 1 penalty for late submission in each month."
+      }
+    }
+
+    "given multiple penalty period in LSPP" should{
```
Comment (LewisMitchell) [inline test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:680]:
```suggestion
    "given multiple penalty period in LSPP" should {
```

File: test/views/components/LateSubmissionPenaltySummaryCardSpec.scala [hunk 10]
```diff
@@ -599,6 +669,27 @@ class LateSubmissionPenaltySummaryCardSpec extends SpecBase with ViewBehaviours
...
+      implicit val doc: Document = asDocument(summaryCardHtml.apply(summaryCardModelWithMultiplePenaltyPeriodLSPP))
+      "show message VAT return submitted earlier in multiple penalty period" in{
+        doc.select("p.govuk-body").text() shouldBe "The VAT Return due on 24 May 2021 was also submitted late. HMRC only applies 1 penalty for late submission in each month."
+      }
+    }
+
+    "given no multiple penalty period in LSP" should{
```
Comment (LewisMitchell) [inline test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:687]:
```suggestion
    "given no multiple penalty period in LSP" should {
```

File: test/views/components/LateSubmissionPenaltySummaryCardSpec.scala [hunk 11]
```diff
@@ -599,6 +669,27 @@ class LateSubmissionPenaltySummaryCardSpec extends SpecBase with ViewBehaviours
...
+    "given multiple penalty period in LSP" should{
+      implicit val doc: Document = asDocument(summaryCardHtml.apply(summaryCardModelWithMultiplePenaltyPeriodLSP))
+      "show message VAT return submitted earlier in multiple penalty period" in{
+        doc.select("p.govuk-body").text() shouldBe "The VAT Return due on 24 May 2021 was also submitted late. HMRC only applies 1 penalty for late submission in each month."
+      }
+    }
+
+    "given multiple penalty period in LSPP" should{
+      implicit val doc: Document = asDocument(summaryCardHtml.apply(summaryCardModelWithMultiplePenaltyPeriodLSPP))
+      "show message VAT return submitted earlier in multiple penalty period" in{
```
Comment (LewisMitchell) [inline test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:682]:
```suggestion
      "show message VAT return submitted earlier in multiple penalty period" in {
```

File: test/views/components/LateSubmissionPenaltySummaryCardSpec.scala [hunk 12]
```diff
@@ -599,6 +669,27 @@ class LateSubmissionPenaltySummaryCardSpec extends SpecBase with ViewBehaviours
...
+        doc.select("p.govuk-body").text() shouldBe "The VAT Return due on 24 May 2021 was also submitted late. HMRC only applies 1 penalty for late submission in each month."
+      }
+    }
+
+    "given no multiple penalty period in LSP" should{
+      implicit val doc: Document = asDocument(summaryCardHtml.apply(summaryCardModelWithFinancialPointAboveThreshold))
+      "show message VAT return submitted earlier in multiple penalty period" in{
```
Comment (LewisMitchell) [inline test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:689]:
```suggestion
      "show message VAT return submitted earlier in multiple penalty period" in {
```

Comment (LewisMitchell) [reply test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:689]: Also I'm not sure this test name makes sense - perhaps we could change this to be:
`"no message relating to multiple penalties in the same period should appear" in {`

File: test/views/components/LateSubmissionPenaltySummaryCardSpec.scala [hunk 13]
```diff
@@ -599,6 +669,27 @@ class LateSubmissionPenaltySummaryCardSpec extends SpecBase with ViewBehaviours
         docWithAppealedPointUnderTribunalRejected.select("dd").get(4).text() shouldBe "Appeal rejected by tax tribunal Read outcome message"
       }
     }
+
+    "given multiple penalty period in LSP" should{
+      implicit val doc: Document = asDocument(summaryCardHtml.apply(summaryCardModelWithMultiplePenaltyPeriodLSP))
+      "show message VAT return submitted earlier in multiple penalty period" in{
```
Comment (LewisMitchell) [inline test/views/components/LateSubmissionPenaltySummaryCardSpec.scala:675]:
```suggestion
      "show message VAT return submitted earlier in multiple penalty period" in {
```

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, just some clean-up & formatting tasks

**Comment by ruthdas03** [comment]
> Looks good, just some clean-up & formatting tasks

Updated with the suggested changes

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**